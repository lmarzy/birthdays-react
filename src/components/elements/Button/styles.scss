.btn {
  display: inline-block;
  padding: $rhythm / 2 $gutter * 2;
  vertical-align: middle;
  font: inherit;
  text-align: center;
  cursor: pointer;
  transition: all 300ms ease-in-out;
}

.primary {
  background: $brand-a;
  color: white;

  &:hover {
    background: darken($brand-a, 10);
  }
}
