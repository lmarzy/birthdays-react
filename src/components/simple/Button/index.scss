@import '../../../assets/css/01-settings/__imports.scss';
@import '../../../assets/css/02-tools/__imports.scss';

.btn {
  display: inline-block;
  padding: $rhythm / 2 $gutter * 2;
  vertical-align: middle;
  font: inherit;
  text-align: center;
  cursor: pointer;
  transition: all 300ms ease-in-out;
}

.primary {
  background: $brand-a;
  color: white;

  &:hover {
    background: darken($brand-a, 10);
  }
}
