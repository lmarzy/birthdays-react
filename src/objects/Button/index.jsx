import React, { PropTypes } from 'react';
import './index.scss';

const Button = props => (
  <button type={props.type} className="btn">{props.text}</button>
);

Button.propTypes = {
  type: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default Button;
