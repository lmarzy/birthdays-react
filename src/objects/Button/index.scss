@import '../../styles/01-settings/__imports.scss';
@import '../../styles/02-tools/__imports.scss';

.btn {
  display: inline-block;
  padding: $rhythm / 2 $gutter * 2;
  vertical-align: middle;
  font: inherit;
  text-align: center;
  cursor: pointer;
  transition: all 300ms ease-in-out;
}

.primary {
  background: $brand-a;
  color: white;

  &:hover {
    background: darken($brand-a, 10);
  }
}
