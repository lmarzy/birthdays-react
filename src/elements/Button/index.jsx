import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scss';

const Button = props => (
  <button type={props.type} className={styles.btn}>
    {props.children}
  </button>
);

Button.propTypes = {
  type: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired,
};

export default Button;
