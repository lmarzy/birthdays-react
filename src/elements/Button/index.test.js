import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Button from './index';

const component = shallow(<Button type="button">Testing</Button>);

describe('<Button />', () => {
  it('Should render a "<Button />" component', () => {
    expect(component).toHaveLength(1);
  });

  it('Should be of type "button"', () => {
    expect(component.type()).toEqual('button');
  });

  it('renders a snapshot', () => {
    const tree = renderer.create(<Button type="button">Testing</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
