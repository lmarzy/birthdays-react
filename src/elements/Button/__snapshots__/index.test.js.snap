// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders a snapshot 1`] = `
<button
  className="btn"
  type="button"
>
  Testing
</button>
`;
